# GeneMark-EP+

_GeneMark-EP and -EP+: automatic eukaryotic gene prediction supported by spliced aligned proteins_

(manuscript submitted for publication, 2019)

Tomas Bruna, Alexandre Lomsadze, Mark Borodovsky

Georgia Institute of Technology, Atlanta, Georgia, USA


# Overview

GeneMark-EP+ is a semi-supervised eukaryotic gene prediction tool which
utilizes protein hints to improve unsupervised parameter estimation and
predictions.

Protein hints are generated by
[ProtHint](https://github.com/gatech-genemark/ProtHint), a fast protein
mapping pipeline which predicts and scores introns, start and stop codons in
the genome of interest from any number of proteins of unknown evolutionary
distance (we recommend to use all eukaryotic proteins in
[OrthDB](https://www.orthodb.org/?page=filelist)).

Due to its semi-supervised nature and ability to incorporate proteins of
any evolutionary distance, GeneMark-EP+ is an optimal tool to predict genes in
a novel genome without the need for a curated training set or a set of
closely related proteins.


# Downloads

### Tools

* GeneMark-EP+ is available for download at http://topaz.gatech.edu/GeneMark/license_download.cgi
* ProtHint is available at https://github.com/gatech-genemark/ProtHint

### Experiments

* Project with experiments and evaluation of GeneMark-EP+ and ProtHint is available at https://github.com/gatech-genemark/GeneMark-EP-experiments.

# Usage example

First run ProtHint to get protein hints (see [ProtHint repository](https://github.com/gatech-genemark/ProtHint) for details about usage and installation)

```bash
prothint.py genome.masked.fasta proteins.fasta --workdir ProtHintDir
```

Run GeneMark-EP+ with hints mapped by ProtHint.

```bash
gmes_petap.pl --EP ProtHintDir/prothint.gff --evidence ProtHintDir/evidence.gff --seq genome.masked.fasta --soft_mask 1000 --verbose
```

# Runtime

Runtime of GeneMark-EP+ is linear with respect to genome size.

ProtHint runtime is linear with respect to both genome size (GeneMark-ES is executed to generate initial genome seeds) and to the number of genes in a genome.

### Examples

Runs were executed on a 8CPU/8GB RAM machine. Genomes were masked for repeats by [RepeatModeler](http://www.repeatmasker.org/RepeatModeler/) and [RepeatMasker](http://www.repeatmasker.org/RMDownload.html). Proteins from species within the same taxonomical genus were excluded in these experiments.

**_Drosophila melanogaster_** (134 Mb and \~14,000 genes) with [OrthoDB _Arthropoda_ ](https://v100.orthodb.org/download/odb10_arthropoda_fasta.tar.gz) target proteins:
* ProtHint: 3h 15m (1h 45m GeneMark-ES, 1h 30m protein mapping)
* GeneMark-EP+: 1h 30m
* Total: **~5 hours**

**_Solanum lycopersicum_** (807 Mb and \~35,000 genes) with [OrthoDB _Plantae_](https://v100.orthodb.org/download/odb10_plants_fasta.tar.gz)  target proteins:
* ProtHint: 9h (5h GeneMark-ES, 4h protein mapping)
* GeneMark-EP+: 4h
* Total: **~13 hours**

